core/Azure.Mcp.Core/src/Services/Azure/Subscription/SubscriptionService.cs

```diff
@@ -47,8 +47,23 @@ public async Task<SubscriptionResource> GetSubscription(string subscription, str
     {
         ValidateRequiredParameters((nameof(subscription), subscription));
 
-        // Get the subscription ID first, whether the input is a name or ID
-        var subscriptionId = await GetSubscriptionId(subscription, tenant, retryPolicy, cancellationToken);
+        string subscriptionId;
+        try
+        {
+            // Get the subscription ID first, whether the input is a name or ID
+            subscriptionId = await GetSubscriptionId(subscription, tenant, retryPolicy, cancellationToken);
+        }
+        catch
+        {
+            // If the subscription lookup failed, try environment variable as fallback
+            var envSubscription = Core.Helpers.EnvironmentHelpers.GetAzureSubscriptionId();
+            if (string.IsNullOrEmpty(envSubscription))
+            {
+                throw; // Re-throw the original exception if no environment variable is available
+            }
+
+            subscriptionId = await GetSubscriptionId(envSubscription, tenant, retryPolicy, cancellationToken);
+        }
```

Comment:
By the time we reach this point, didn't we already go through the logic in `CommandHelper` which checks for `--subscription` being passed, and if it wasn't falling back to the environment variable `AZURE_SUBSCRIPTION_ID`? So, wouldn't this be more confusing as if I pass `--subscription invalid` but have `AZURE_SUBSCRIPTION_ID=valid` I now have a complicated to understand flow with my `--subscription` input being ignored and `--subscription valid` was used.